app.js

```diff
@@ -1,21 +1,31 @@
 const { getTopics } = require("./controllers/topics.controllers")
 const { getEndpoints } = require("./controllers/api.controllers")
+const { getArticleById } = require("./controllers/articles.controllers")
 
 const express = require("express")
 
 const app = express()
 
-app.use(express.json())
-
 app.get("/api/topics", getTopics)
 
 app.get("/api", getEndpoints)
 
+app.get("/api/articles/:article_id", getArticleById)
+
+app.all('*', (req, res) => {
+    res.status(404).send({msg: "The request path does not exist"})
+})
+
 app.use((err, req, res, next) => {
+    if (err.msg) {
+        res.status(err.status).send({msg: err.msg})
+    } else { 
     res.status(500).send('Server Error! :(')
```

Comment:
This should really be a separate error handling middleware. Errors are passed to the next error handling middleware by invoking next with the error (eg. next(error)). You can just add another app.use for this functionality. Just make sure server error handling comes last in the error handling middleware chain.